src/pages/VaccineModulePage.js

```diff
@@ -0,0 +1,264 @@
+  HAZARD_RED,
+} from '../globalStyles';
+
+const CHERVON_ICON_STYLE = { size: 18, color: SUSSOL_ORANGE };
+const BREACH_ICON_STYLE = { size: 25, color: HAZARD_RED };
+
+// TODO navigation for menu buttons
+// TODO Localise all strings
+
+export class VaccineModulePage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { fridges: [], selectedFridgeCode: null };
+  }
+
+  componentWillMount() {
+    const { database } = this.props;
+    const fridges = database.objects('Location').filter(({ isFridge }) => isFridge);
+    const selectedFridgeCode = fridges.length > 0 && fridges[0].code;
+    this.setState({ fridges, selectedFridgeCode });
+  }
+
+  /* Helper to render menuButton */
+  renderMenuButton = buttonProps => (
+    <Button
+      style={[globalStyles.menuButton, localStyles.menuButton]}
+      textStyle={globalStyles.menuButtonText}
+      {...buttonProps}
+    />
+  );
+
+  renderIcon = (iconName, iconStyle = CHERVON_ICON_STYLE) => (
+    <Icon style={{ margin: 5 }} name={iconName} {...iconStyle} />
+  );
+
+  /* Render fridge name and chevron-down icon if icon not selected fridge */
+  renderFridgeName = (fridge, isFridgeSNotelected) => {
+    const onPress = () => this.setState({ selectedFridgeCode: fridge.code });
+
+    const { fridgeInfoSectionStyle, fridgeNameTextStyle } = localStyles;
+
+    return (
+      <TouchableOpacity style={fridgeInfoSectionStyle} onPress={onPress}>
+        {!isFridgeSNotelected && this.renderIcon('chevron-down')}
+        <Text style={fridgeNameTextStyle}>{fridge.description}</Text>
+      </TouchableOpacity>
+    );
+  };
+
+  /* Temperature with celcius symbol based no passed fontSize */
+  renderTemperature = (fontSize, temperature) => {
+    const { greyTextStyleLarge } = localStyles;
+    return (
+      <View style={{ flexDirection: 'row' }}>
+        <Text style={[greyTextStyleLarge, { fontSize }]}>{temperature}</Text>
+        <Text style={[greyTextStyleLarge, { fontSize: fontSize * 0.7 }]}>℃</Text>
+      </View>
+    );
+  };
+
+  /* Render 'Breach: {num} Exposure: {fromTemp} to {toTemp} */
+  renderFridgeExtraInfo = fridge => {
+    const numberOfBreaches = fridge.getNumberOfBreaches();
+    const hasBreaches = numberOfBreaches > 0;
+    const { minTemperature, maxTemperature } = fridge.getTemperatureExposure();
+
+    const { greyTextStyleSmall, greyTextStyleLarge, fridgeInfoSectionStyle } = localStyles;
+    const extraSectionStyle = { flexDirection: 'row', justifySelf: 'flex-end', marginRight: 10 };
+
+    return (
+      <View style={[fridgeInfoSectionStyle, extraSectionStyle]}>
+        {hasBreaches && (
+          <View style={[fridgeInfoSectionStyle, { marginRight: 10 }]}>
+            <Text style={greyTextStyleSmall}>Breaches:</Text>
+            <Text style={greyTextStyleLarge}>{numberOfBreaches}</Text>
+          </View>
+        )}
+        <Text style={greyTextStyleSmall}>Exposure:</Text>
+        {this.renderTemperature(20, minTemperature)}
```

Comment:
give `renderTemperature` an optional second temp, if two are supplied return X to Y

Reply:
referenced at the bottom of pr